src/Npgsql/SqlQueryParser.cs

```diff
@@ -433,5 +461,28 @@ static bool IsIdentifier(char ch)
 
         static bool IsParamNameChar(char ch)
             => char.IsLetterOrDigit(ch) || ch == '_' || ch == '.';  // why dot??
+
+        static int IndexOf(ReadOnlySpan<char> span, ReadOnlySpan<char> subSpan, int startPosition)
+        {
+            var result = span.Slice(startPosition).IndexOf(subSpan);
+            if (result != -1)
+                result += startPosition; // If the substring is found adjust the result to be relative to the entire span
+            return result;
+        }
+
+#if (NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 ||NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6 || NETSTANDARD2_0 || NET452)
```

Comment:
I don't think there's any need to have all these netstandard versions - it should be enough to only list those we actually target. If we ever start targeting one of these without adding it to this list, it won't compile so we'll know right away.

Regardless, a crazy idea: rather than having special static method, why not simply define the missing Span overloads as extensions methods for old TFMs? This way we only have to #if for older TFMs, and the code looks just like it would on newer TFMs.

Reply:
> why not simply define the missing Span overloads as extensions methods
Extension methods on StringBuilder? If we can append a ReadOnlySpan then most of what I wrote goes away.

Reply:
I'm not sure why not... Of course this would be an internal class - ideally it would have been scoped only to the SqlQueryParser class, but IIRC extension methods can only be defined in top-level classes. Even so, I don't really see any actual harm in having these extension methods within Npgsql for old TFMs...